pkg/realizer/runnable/realizer.go

```diff
@@ -139,6 +142,10 @@ func (r *runnableRealizer) Realize(ctx context.Context, runnable *v1alpha1.Runna
 
 	if outputSource != nil {
 		log.V(logger.DEBUG).Info("retrieved output from stamped object", "stamped object", outputSource)
+		if !reflect.DeepEqual(runnable.Status.Outputs, map[string]apiextensionsv1.JSON(outputs)) {
+			rec := events.FromContextOrDie(ctx)
+			rec.ResourceEventf(events.NormalType, events.ResourceOutputChangedReason, "Resource [%Q] outputs changed to %s", stampedObject, outputs)
```

Comment:
uh I think having the full output might lead to too much polution 🤔

e.g., consider the following supply chain where we pass config from one resource to another:

```yaml
apiVersion: carto.run/v1alpha1
kind: ClusterSupplyChain
metadata:
  name: supplychain
spec:
  selector:
    foo: bar

  resources:
    - name: config-provider           # comes up with a config, making
      templateRef:                    # available to anyone who wants it
        kind: ClusterConfigTemplate
        name: config-provider

    - name: config-consumer
      templateRef:
        kind: ClusterTemplate
        name: config-consumer
      configs:
        - name: config
          resource: config-provider   # consumes that config from the provider
```

with the templates looking like

```yaml
---
apiVersion: carto.run/v1alpha1
kind: ClusterConfigTemplate
metadata:
  name: config-provider
spec:
  configPath: .data

  template:
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: $(workload.metadata.name)$-provider
    data:
      foo: bar
      caz: zaz
      baz: |-
        multiline string
        that is maybe
        a little too much

        right

---
apiVersion: carto.run/v1alpha1
kind: ClusterTemplate
metadata:
  name: config-consumer
spec:
  template:
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: $(workload.metadata.name)$-consumer
    data: $(config)$
```

then creating a workload that matches the selector as described in the supply
chain


```yaml
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: workload
  labels:
    foo: bar
spec: {}
```

we end up with

```
Events:
  Type    Reason                 Age                From      Message
  ----    ------                 ----               ----      -------
  Normal  StampedObjectApplied   60s                Workload  Created object [configmaps/workload-provider]
  Normal  StampedObjectApplied   60s                Workload  Created object [configmaps/workload-consumer]
  Normal  ResourceOutputChanged  54s (x3 over 60s)  Workload  Resource [configmaps/workload-provider] outputs changed to [config:baz: |-
    multiline string
    that is maybe
    a little too much

    right
caz: zaz
foo: bar
]
  Normal  StampedObjectApplied  49s (x4 over 57s)  Workload  Patched object [configmaps/workload-provider]
  Normal  StampedObjectApplied  49s (x4 over 57s)  Workload  Patched object [configmaps/workload-consumer]
```

Reply:
Yes, and that's before we even consider whether this information is sensitive or not. The RFC does not call for the output values to be in the event, perhaps I should remove this for now and await further feedback? WDYT?

It will give us some time to consider how we might deal with sensitive data as well -- it may be that we will treat this the same as #384

Reply:
👍 sgtm removing for now 😁